sky/clouds/ibm.py

```diff
@@ -0,0 +1,454 @@
+        else:
+            regions = service_catalog.get_region_zones_for_instance_type(
+                instance_type, use_spot, 'ibm')
+
+        if region is not None:
+            regions = [r for r in regions if r.name == region]
+        if zone is not None:
+            for r in regions:
+                r.set_zones([z for z in r.zones if z.name == zone])
+            regions = [r for r in regions if r.zones]
+        return regions
+
+    @classmethod
+    def region_zones_provision_loop(
+        cls,
+        *,
+        instance_type: Optional[str] = None,
+        accelerators: Optional[Dict[str, int]] = None,
+        use_spot: Optional[bool] = False,
+    ) -> Iterator[Tuple[Region, List[Zone]]]:
+        """Loops over (region, zones) to retry for provisioning.
+
+        returning a single zone list with its region,
+        since ibm cloud currently doesn't
+        support retries for list of zones.
+
+        Args:
+            instance_type: The instance type to provision.
+            accelerators: The accelerators to provision.
+            use_spot: Whether to use spot instances.
+        """
+        del accelerators  # unused
+        del use_spot  # unsupported
+
+        if instance_type is None:
+            # fallback to manually specified region/zones
+            regions = cls.regions()
+        else:
+            regions = service_catalog.get_region_zones_for_instance_type(
+                instance_type, False, 'ibm')
+
+        for region in regions:
+            for zone in region.zones:
+                yield region, [zone]
+
+    @classmethod
+    def get_zone_shell_cmd(cls) -> Optional[str]:
+        return None
+
+    def instance_type_to_hourly_cost(self,
+                                     instance_type: str,
+                                     use_spot: bool,
+                                     region: Optional[str] = None,
+                                     zone: Optional[str] = None) -> float:
+        # Currently doesn't support spot instances, hence use_spot set to False.
+        del use_spot
+        return service_catalog.get_hourly_cost(instance_type,
+                                               use_spot=False,
+                                               region=region,
+                                               zone=zone,
+                                               clouds='ibm')
+
+    def accelerators_to_hourly_cost(self,
+                                    accelerators: Dict[str, int],
+                                    use_spot: bool,
+                                    region: Optional[str] = None,
+                                    zone: Optional[str] = None) -> float:
+        del accelerators, use_spot, region, zone  # unused
+        # Currently Isn't implemented in the same manner by aws and azure.
+        return 0
+
+    def get_egress_cost(self, num_gigabytes):
+        """Returns the egress cost. Currently true for us-south, i.e. Dallas.
```

Comment:
Of course, it is only used to get a rough estimation.
In a similar manner to aws's implementation these values are hard-coded and represent a single region. 
perhaps in the future it could fetch relevant data via a web-crawler.